lib/MusicBrainz/Server/Data/Utils.pm

```diff
@@ -397,17 +398,23 @@ my $noncharacter_pattern = '\x{FDD0}-\x{FDEF}\x{FFFE}\x{FFFF}';
 
 sub remove_invalid_characters {
     shift
-    # trim XML-invalid characters
+    # trim XML-invalid characters, among them:
+    # - Other, surrogate (which are UTF-16, not valid UTF-8)
     =~ s/[^\x09\x0A\x0D\x20-\x{D7FF}\x{E000}-\x{FFFD}\x{10000}-\x{10FFFF}]//gor
     # trim other undesirable characters:
-    # - zwsp
-    # - shy
     # - bom
-    # - Other, control
-    # - Other, surrogate
     # - Supplementary private use areas
     # - Noncharacters
-    =~ s/[\x{200B}\x{00AD}\x{FEFF}\p{Cc}\x{F0000}-\x{FFFFF}\x{100000}-\x{10FFFF}${noncharacter_pattern}]//gr
+    =~ s/[\x{FEFF}\x{F0000}-\x{FFFFF}\x{100000}-\x{10FFFF}${noncharacter_pattern}]//gr
+}
+
+sub remove_lineformatting_characters {
+    shift
+    # trim other undesirable characters:
```

Comment:
I'd suggest making this comment more clear - I expect this is not "other undesirable characters" but "line formatting characters"? :)

Reply:
Specified “lasting line-formatting characters” in commit https://github.com/metabrainz/musicbrainz-server/pull/1944/commits/62348af4bfaa6d665ef5a76e7d945c53199ea37f because some have been removed already.